Jobs.Fetcher.Twitter/Helpers/DbReader.cs

```diff
@@ -18,49 +18,98 @@ namespace Jobs.Fetcher.Twitter.Helpers {
 
     public static class DbReader {
 
-        public static User GetUserByUsername(string username, DataLakeTwitterDataContext dbContext) {
-            return dbContext.Users.FirstOrDefault(u => u.Username == username);
+        public static User GetUserByUsername(string username, DataLakeTwitterDataContext dbContext, Logger logger) {
+            try {
+                return dbContext.Users.FirstOrDefault(u => u.Username == username);
+            } catch (Exception e) {
+                logger.Error($"Error when looking for User {username} in the database");
+                logger.Verbose($"Error: {e}");
+                return null;
+            }
         }
 
-        public static IEnumerable<LocalAdsAccount> GetAdsAccounts(string username, DataLakeTwitterAdsContext dbContext) {
-            return dbContext.AdsAccounts.Where(a => a.Username == username).ToList();
+        public static IEnumerable<LocalAdsAccount> GetAdsAccounts(string username, DataLakeTwitterAdsContext dbContext, Logger logger) {
+            try {
+                return dbContext.AdsAccounts.Where(a => a.Username == username).ToList();
+            } catch (Exception e) {
+                logger.Error($"Error when looking for AdsAccounts for {username} in the database");
+                logger.Verbose($"Error: {e}");
+                return new List<LocalAdsAccount>();
+            }
         }
 
-        public static IEnumerable<string> GetAdsAccountIds(string username, DataLakeTwitterAdsContext dbContext) {
-            return dbContext.AdsAccounts.Where(a => a.Username == username).Select(a => a.Id).ToList();
+        public static IEnumerable<string> GetAdsAccountIds(string username, DataLakeTwitterAdsContext dbContext, Logger logger) {
+            try {
+                return dbContext.AdsAccounts.Where(a => a.Username == username).Select(a => a.Id).ToList();
```

Comment:
I am also not sure about this here. Maybe this check should be in the caller function? Like, if there are no accounts to retrieve, you send no accounts (an empty list), this is not an exception, it is the expected behavior. But, in the caller function, if it finds out that the username we are looking for does not have any ad accounts, then we cannot go further with what we were doing